panic found in dispatcher

Need to find out why the tableTrigger dispatcher cannot be created successfully in L170:
https://github.com/pingcap/ticdc/blob/bdef9abbd573027817ad6156ecc75588fc764cff/downstreamadapter/dispatchermanager/event_dispatcher_manager.go#L169-L172

```log
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x463598a]

goroutine 674 [running]:
[github.com/pingcap/ticdc/downstreamadapter/dispatcher.(*Dispatcher).GetStartTs(...)](http://github.com/pingcap/ticdc/downstreamadapter/dispatcher.(*Dispatcher).GetStartTs(...))
        [github.com/pingcap/ticdc/downstreamadapter/dispatcher/dispatcher.go:456](http://github.com/pingcap/ticdc/downstreamadapter/dispatcher/dispatcher.go:456)
[github.com/pingcap/ticdc/downstreamadapter/dispatchermanager.NewEventDispatcherManager](http://github.com/pingcap/ticdc/downstreamadapter/dispatchermanager.NewEventDispatcherManager)({{0x56e35bc?, 0x467?}, {0xc0185a9a98?, 0x4a29c40?}}, 0xc0392de000, 0xc010f0de60, 0x64b9cb2ee140001, {0xc000079f80, 0x24})
        [github.com/pingcap/ticdc/downstreamadapter/dispatchermanager/event_dispatcher_manager.go:171](http://github.com/pingcap/ticdc/downstreamadapter/dispatchermanager/event_dispatcher_manager.go:171) +0x8ea
[github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator.(*DispatcherOrchestrator).handleAddDispatcherManager](http://github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator.(*DispatcherOrchestrator).handleAddDispatcherManager)(0xc00352cd80, {0xc000079f80, 0x24}, 0xc0034ff9c0)
        [github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator/dispatcher_orchestrator.go:73](http://github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator/dispatcher_orchestrator.go:73) +0x335
[github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator.(*DispatcherOrchestrator).RecvMaintainerRequest](http://github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator.(*DispatcherOrchestrator).RecvMaintainerRequest)(0xc0011ae8c0?, {0x61e04e0?, 0xc004e86010?}, 0xc00301fc70?)
        [github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator/dispatcher_orchestrator.go:53](http://github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator/dispatcher_orchestrator.go:53) +0x94
[github.com/pingcap/ticdc/pkg/messaging.(*router).runDispatch.func1()](http://github.com/pingcap/ticdc/pkg/messaging.(*router).runDispatch.func1())
        [github.com/pingcap/ticdc/pkg/messaging/router.go:53](http://github.com/pingcap/ticdc/pkg/messaging/router.go:53) +0x1d4
created by [github.com/pingcap/ticdc/pkg/messaging.(*router).runDispatch](http://github.com/pingcap/ticdc/pkg/messaging.(*router).runDispatch) in goroutine 1
        [github.com/pingcap/ticdc/pkg/messaging/router.go:38](http://github.com/pingcap/ticdc/pkg/messaging/router.go:38) +0xb4
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x463598a]

goroutine 649 [running]:
[github.com/pingcap/ticdc/downstreamadapter/dispatcher.(*Dispatcher).GetStartTs(...)](http://github.com/pingcap/ticdc/downstreamadapter/dispatcher.(*Dispatcher).GetStartTs(...))
        [github.com/pingcap/ticdc/downstreamadapter/dispatcher/dispatcher.go:456](http://github.com/pingcap/ticdc/downstreamadapter/dispatcher/dispatcher.go:456)
[github.com/pingcap/ticdc/downstreamadapter/dispatchermanager.NewEventDispatcherManager](http://github.com/pingcap/ticdc/downstreamadapter/dispatchermanager.NewEventDispatcherManager)({{0x56e35bc?, 0x467?}, {0xc01bea6e3a?, 0x4a29c40?}}, 0xc003ea6200, 0xc008b25ae0, 0x64b9cb2ee140001, {0xc0035da450, 0x24})
        [github.com/pingcap/ticdc/downstreamadapter/dispatchermanager/event_dispatcher_manager.go:171](http://github.com/pingcap/ticdc/downstreamadapter/dispatchermanager/event_dispatcher_manager.go:171) +0x8ea
[github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator.(*DispatcherOrchestrator).handleAddDispatcherManager](http://github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator.(*DispatcherOrchestrator).handleAddDispatcherManager)(0xc00358a828, {0xc0035da450, 0x24}, 0xc00709a240)
        [github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator/dispatcher_orchestrator.go:73](http://github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator/dispatcher_orchestrator.go:73) +0x335
[github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator.(*DispatcherOrchestrator).RecvMaintainerRequest](http://github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator.(*DispatcherOrchestrator).RecvMaintainerRequest)(0xc0008fa820?, {0x61e04e0?, 0xc0011581b0?}, 0xc0035ae8f0?)
        [github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator/dispatcher_orchestrator.go:53](http://github.com/pingcap/ticdc/downstreamadapter/dispatcherorchestrator/dispatcher_orchestrator.go:53) +0x94
[github.com/pingcap/ticdc/pkg/messaging.(*router).runDispatch.func1()](http://github.com/pingcap/ticdc/pkg/messaging.(*router).runDispatch.func1())
        [github.com/pingcap/ticdc/pkg/messaging/router.go:53](http://github.com/pingcap/ticdc/pkg/messaging/router.go:53) +0x1d4
created by [github.com/pingcap/ticdc/pkg/messaging.(*router).runDispatch](http://github.com/pingcap/ticdc/pkg/messaging.(*router).runDispatch) in goroutine 1
        [github.com/pingcap/ticdc/pkg/messaging/router.go:38](http://github.com/pingcap/ticdc/pkg/messaging/router.go:38) +0xb4
```

--- a/downstreamadapter/dispatchermanager/event_dispatcher_manager.go
+++ b/downstreamadapter/dispatchermanager/event_dispatcher_manager.go
@@ -27,6 +27,7 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/tikv/client-go/v2/oracle"
 
+	"github.com/pingcap/errors"
 	"github.com/pingcap/ticdc/downstreamadapter/dispatcher"
 	"github.com/pingcap/ticdc/downstreamadapter/eventcollector"
 	"github.com/pingcap/ticdc/downstreamadapter/sink"
@@ -166,7 +167,11 @@ func NewEventDispatcherManager(changefeedID common.ChangeFeedID,
 
 	// create tableTriggerEventDispatcher if it is not nil
 	if tableTriggerEventDispatcherID != nil {
-		manager.NewDispatcher(common.NewDispatcherIDFromPB(tableTriggerEventDispatcherID), heartbeatpb.DDLSpan, startTs, 0)
+		_, err := manager.NewDispatcher(common.NewDispatcherIDFromPB(tableTriggerEventDispatcherID), heartbeatpb.DDLSpan, startTs, 0)
+		if err != nil {
+			log.Error("Create table trigger event dispatcher failed", zap.Error(err))
+			return nil, 0, apperror.ErrCreateEventDispatcherManagerFailed.Wrap(err).GenWithStackByArgs("create table trigger event dispatcher failed")
+		}
 		return manager, manager.tableTriggerEventDispatcher.GetStartTs(), nil
 	}
 
@@ -234,10 +239,10 @@ func (e *EventDispatcherManager) close(remove bool) {
 	log.Info("event dispatcher manager closed", zap.Stringer("changefeedID", e.changefeedID))
 }
 
-func (e *EventDispatcherManager) NewDispatcher(id common.DispatcherID, tableSpan *heartbeatpb.TableSpan, startTs uint64, schemaID int64) *dispatcher.Dispatcher {
+func (e *EventDispatcherManager) NewDispatcher(id common.DispatcherID, tableSpan *heartbeatpb.TableSpan, startTs uint64, schemaID int64) (*dispatcher.Dispatcher, error) {
 	start := time.Now()
 	if _, ok := e.dispatcherMap.Get(id); ok {
-		return nil
+		return nil, nil
 	}
 
 	// If downstream is mysql-class, we need to check whether startTs we should actually use as the begin startTs
@@ -246,8 +251,7 @@ func (e *EventDispatcherManager) NewDispatcher(id common.DispatcherID, tableSpan
 	// and choose max(ddl-ts, startTs) as the begin startTs.
 	newStartTs, err := e.sink.CheckStartTs(tableSpan.TableID, startTs)
 	if err != nil {
-		log.Error("check start ts failed", zap.Error(err))
-		return nil
+		return nil, errors.Trace(err)
 	}
 
 	// this table is dropped, skip it and return removed status to maintainer
@@ -257,7 +261,7 @@ func (e *EventDispatcherManager) NewDispatcher(id common.DispatcherID, tableSpan
 			ComponentStatus: heartbeatpb.ComponentState_Removed,
 		}
 		log.Info("this table is dropped, skip it and return removed status to maintainer")
-		return nil
+		return nil, nil
 	}
 
 	syncPointInfo := syncpoint.SyncPointInfo{
@@ -312,7 +316,7 @@ func (e *EventDispatcherManager) NewDispatcher(id common.DispatcherID, tableSpan
 		zap.Int64("cost(ns)", time.Since(start).Nanoseconds()), zap.Time("start", start))
 	e.metricCreateDispatcherDuration.Observe(float64(time.Since(start).Seconds()))
 
-	return d
+	return d, nil
 }
 
 func (e *EventDispatcherManager) CollectBlockStatusRequest(ctx context.Context) {

--- a/pkg/apperror/util.go
+++ b/pkg/apperror/util.go
@@ -25,7 +25,7 @@ func IsTableNotExistsErr(err error) bool {
 	}
 
 	errCode := errors.ErrCode(mysqlErr.Number)
-	return errCode == infoschema.ErrTableNotExists.Code()
+	return errCode == infoschema.ErrTableNotExists.Code() || errCode == infoschema.ErrDatabaseNotExists.Code()
 }
 
 // IsIgnorableMySQLDDLError is used to check what error can be ignored

